engine/utils.hpp

```diff
@@ -6,21 +6,32 @@
 
 #define XXH_INLINE_ALL
 
-#include <assert.h>
-#include <atomic>
+#include <cassert>
 #include <cstdint>
-#include <emmintrin.h>
+
+#include <atomic>
+#include <exception>
 #include <random>
-#include <smmintrin.h>
 #include <string>
+#include <vector>
+
 #include <sys/stat.h>
 #include <sys/time.h>
-#include <vector>
+
+#include <emmintrin.h>
+#include <smmintrin.h>
 
 #include "kvdk/namespace.hpp"
 #include "libpmemobj++/string_view.hpp"
 #include "xxhash.h"
 
+#define kvdk_assert(cond, msg)                                                 \
```

Comment:
this is still assert + throw, I think it should only throw while DEBUG_LEVEL is not 0

Reply:
added DEBUG_LEVEL flag to CMakeLists.txt and Release build will simply assert, aka, do nothing.
DEBUG_LEVEL is independent of NDEBUG flag or DEBUG flag so when DEBUG_LEVEL = 0 kvdk_assert is assert instead of nothing.